dhall/src/Dhall.hs

```diff
@@ -900,6 +909,103 @@ instance (Interpret a, Interpret b) => Interpret (a, b)
 auto :: Interpret a => Type a
 auto = autoWith defaultInterpretOptions
 
+newtype DhallFix f t = DhallFix { unDhallFix :: t }
+newtype FixVar = FixVar { unFixVar :: (Expr Src X) }
+
+-- This is a helper instance used from implementation of Interpret for DhallFix
+instance Interpret FixVar where
+    autoWith _ = Type {..}
+        where
+        extract (App _ e0) = pure (FixVar e0)
```

Comment:
If you α-normalize before extraction (see comment below), then you can add an additional check that the correct bound variable is used here:

```suggestion
        extract (App (Var (V "_" 0)) e0) = pure (FixVar e0)
```

Reply:
Yeah so my thinking here was, is this actually needed? If this type checks it should be a correct expression anyway. But now I'm not 100%. I can imagine there being a scenario somewhat like this:

```dhall
  [..]
    λ(e : Type)
  → λ(fix : ExprF e → e)
  → let E = ExprF e
    let g = λ(_ : ExprF e) → fix (E.LitF { _1 = 1 })
    let Mul = λ(a : e) → λ(b : e) → fix (E.MulF { _1 = a, _2 = b })
    in
  [..]
```

And then this whole expression is used in some lambda, gets mingled and renamed and somehow `g` that has the same type as `fix` gets used in `Mul`, or something like that?
So ideally I would like an example of such expression so I could add it to the test suite and make sure that I'm rejecting it correctly.

So my question here is, is this:
  1. You know it's possible and can give me an example so I should do it.
  2. You know it's likely but cannot give me a proper example from the top of your had so I should do it anyway.
  3. You're not sure but better safe than sorry so I should do it.
  4. It's not actually possible.

Reply:
I believe it's "4" but it makes the code more self-documenting